community/ndp/runtime/src/main/java/org/neo4j/ndp/runtime/NDPRuntime.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.ndp.runtime;
+
+/**
+ * The environment incoming user work is executed in.
+ */
+public interface NDPRuntime
```

Comment:
I'm super keen to get help with naming here, this is one of those classic things that end up getting called "manager". This interface is the surface that various transports forward incoming messages to, and internally it contains the code that defines the runtime behavior and functionality (eg. the state machine and hot-path runtime code for user sessions). 

I guess we could call it `MessageProcessor` or `SessionEngine`.. we called it `UserEnvironment` for a while, but I didn't feel that was very helpful.. not that "runtime" is much better. Hmm, maybe a name with more focus on its role as a "session manager"..